pgjdbc/src/test/java/org/postgresql/test/jdbc2/DatabaseMetaDataTest.java

```diff
@@ -1961,6 +1961,54 @@ void functionColumns(BinaryMode binaryMode) throws SQLException {
     rs.close();
   }
 
+  @MethodSource("data")
+  @ParameterizedTest(name = "binary = {0}")
+  void charOctetLengthColumnTest(BinaryMode binaryMode) throws SQLException {
+    Assumptions.assumeTrue(TestUtil.haveMinimumServerVersion(con, ServerVersion.v9_2));
+    TestUtil.createTable(con, "char_octet_test",
+        "intarray integer[], mytext varchar(100), mychar char(10), mybinary bytea, " +
+            "mynchar char(20), mynvarchar varchar(50), " +
+            "myclob text, myblob bytea, mylongvarchar text");
+
+    DatabaseMetaData dbmd = con.getMetaData();
+
+    assertCharOctetLength(dbmd, "intarray", true, 0);
+    assertCharOctetLength(dbmd, "mytext", false, 100);
+    assertCharOctetLength(dbmd, "mychar", false, 10);
+    assertCharOctetLength(dbmd, "mybinary", false, -1);
+    assertCharOctetLength(dbmd, "mynchar", false, 20);
+    assertCharOctetLength(dbmd, "mynvarchar", false, 50);
+    assertCharOctetLength(dbmd, "myclob", false, -1);
+    assertCharOctetLength(dbmd, "myblob", false, -1);
+    assertCharOctetLength(dbmd, "mylongvarchar", false, -1);
+
+    TestUtil.dropTable(con, "char_octet_test");
+  }
+
+  private void assertCharOctetLength(DatabaseMetaData dbmd, String columnName, boolean expectNull, int expectedLengthOrCompare)
+      throws SQLException {
+    try (ResultSet rs = dbmd.getColumns(null, null, "char_octet_test", columnName)) {
+      assertTrue(rs.next(), "No row returned for column: " + columnName);
+      assertEquals(columnName, rs.getString("COLUMN_NAME"));
+
+      int charOctetLength = rs.getInt("CHAR_OCTET_LENGTH");
+
+      if (expectNull) {
+        assertTrue(rs.wasNull(), "CHAR_OCTET_LENGTH should be null for column: " + columnName);
+      } else {
+        assertFalse(rs.wasNull(), "CHAR_OCTET_LENGTH should not be null for column: " + columnName);
```

Comment:
The failure would maks the root cause. In most of the cases, I want to know the exact expected/actual value rather than just the fact "it should not be null".

Have you considered something like the following?

```
Integer actual = rs.wasNull() ? null : rs.getInt("CHAR_OCTET_LENGTH");
assertEquals(expected, actual, ...);
```